satpy/readers/nwcsaf_nc.py

```diff
@@ -267,32 +268,29 @@ def get_area_def(self, dsid):
 
         nlines, ncols = self.nc[dsid['name']].shape
 
-        area = self._ensure_area_def_is_in_meters(
-            get_area_def('some_area_name',
-                         "On-the-fly area",
-                         'geosmsg',
-                         proj_str,
-                         ncols,
-                         nlines,
-                         area_extent))
+        crs, area_extent = self._ensure_crs_extents_in_meters(proj_str, area_extent)
+        area = AreaDefinition('some_area_name',
+                              "On-the-fly area",
+                              'geosmsg',
+                              crs,
+                              ncols,
+                              nlines,
+                              area_extent)
 
         return area
 
     @staticmethod
-    def _ensure_area_def_is_in_meters(area_definition):
+    def _ensure_crs_extents_in_meters(proj_str, area_extent):
         """Fix units in Earth shape, satellite altitude and 'units' attribute."""
-        if area_definition.proj_dict["units"] == "km":
-            proj_dict = area_definition.proj_dict.copy()
-            from pyresample.geometry import AreaDefinition
-
+        from pyresample.utils import proj4_str_to_dict
+        proj_dict = proj4_str_to_dict(proj_str)
+        if proj_dict.get('units') == 'km':
             proj_dict["units"] = "m"
             proj_dict["a"] *= 1000.
+            proj_dict["b"] *= 1000.
```

Comment:
Hmm. Wonder why this was missing earlier. Maybe wrap it in `try/except KeyError` if the `'b'` was missing in the data and implied `a == b` and a spherical Earth. Would need to check the data, which I don't have at hand.

Reply:
Ok I was going to wrap it in an if statement. There is also a possibility of pyproj/PROJ replacing it with an `R` for a spheroid (representing both radii). I'll add both.

Reply:
Done.